### PR TITLE
Distress.

### PR DESCRIPTION
**Pull request by David-Stormwell**
## About The Pull Request

If no one else works on it I will.

## Why It's Good For The Game

Remaking the old but better gamemode.
It'll have to be polished quite some.

## Changelog
:cl:
add: Re-added old Distress gamemode.
/:cl:

### CODE REVIEW

**Comment by Naaanii** [comment]
Bless.

**Comment by BraveMole** [comment]
give me back my blue color

**Comment by MiniMeatwad** [comment]
Nice.

**Comment by David-Stormwell** [comment]
Converting to draft in the meantime, probably gonna take a little while to work on this.

**Comment by David-Stormwell** [comment]
> 
> 
> give me back my blue color

No.

File: code/datums/gamemodes/hunt.dm [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+	//Automated bioscan / Queen Mother message
+	if(world.time > bioscan_current_interval)
+		announce_bioscans()
+		var/total[] = count_humans_and_xenos(count_flags = COUNT_IGNORE_XENO_SPECIAL_AREA)
+		var/marines = total[1]
+		var/xenos = total[2]
+		var/bioscan_scaling_factor = xenos / max(marines, 1)
+		bioscan_scaling_factor = max(bioscan_scaling_factor, 0.25)
+		bioscan_scaling_factor = min(bioscan_scaling_factor, 1.5)
+		bioscan_current_interval += bioscan_ongoing_interval * bioscan_scaling_factor
```
Comment (SplinterGP) [inline code/datums/gamemodes/hunt.dm:89]: Move this and all bioscan code to infestation typepath, the parent of this one and distress as both have the bioscan, and remove bioscan code from distress

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/datums/gamemodes/hunt.dm [hunk 2]
```diff
@@ -0,0 +1,263 @@
+/datum/game_mode/infestation/hunt
+	name = "Hunt party"
```
Comment (XSlayer300) [inline code/datums/gamemodes/hunt.dm:2]:
```suggestion
	name = "Hunting Party"
```

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 3]
```diff
@@ -980,6 +981,51 @@
 	succeed_activate()
 	add_cooldown()
 
+////////////////////
+/// Build hunt den
+///////////////////
+/datum/action/xeno_action/activable/build_hunt_den
+	name = "Create hunt den"
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:988]:
```suggestion
	name = "Create Hunting Den"
```

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 4]
```diff
@@ -980,6 +981,51 @@
...
+	action_icon_state = "resin_silo"
+	mechanics_text = "Creates a new hunt den, using 3 headbited bodies"
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:990]:
```suggestion
	mechanics_text = "Create a new hunting den, using 3 headbiten corpses."
```

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 5]
```diff
@@ -980,6 +981,51 @@
...
+	var/list/mob/living/valid_mobs = list()
+	for(var/thing in get_turf(A))
+		if(!ishuman(thing))
+			continue
+		var/mob/living/turf_mob = thing
+		if(turf_mob.stat == DEAD && turf_mob.chestburst == 0)
+			valid_mobs += turf_mob
+
+	if(length(valid_mobs) < required_mobs)
+		to_chat(owner, "<span class='warning'>There are not enough dead bodies, you need [required_mobs] bodies for a silo!</span>")
```
Comment (XSlayer300) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1011]:
```suggestion
		to_chat(owner, "<span class='warning'>There are not enough dead bodies, we need [required_mobs] bodies for a silo!</span>")
```

**Comment by David-Stormwell** [comment]
Closing if Bravemole goes on making his.

**Comment by David-Stormwell** [comment]
Now closing as Bravemole's has gotten to the TM stage.

**Pull request closed without merging**